src/main/java/com/synopsys/integration/alert/common/descriptor/config/field/ConfigField.java

```diff
@@ -23,38 +23,81 @@
 
-    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final String subGroup) {
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final String subGroup,
+        final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
         this.key = key;
         this.label = label;
         this.type = type;
         this.required = required;
         this.sensitive = sensitive;
         this.group = group;
         this.subGroup = subGroup;
+        this.validationFunction = validationFunction;
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group) {
-        this(key, label, type, required, sensitive, group, "");
+        this(key, label, type, required, sensitive, group, "", NO_VALIDATION);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final FieldGroup group, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, group, "", validationFunction);
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final String subGroup) {
-        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup);
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup, NO_VALIDATION);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final String subGroup, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, subGroup, validationFunction);
     }
 
     public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive) {
-        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "");
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "", NO_VALIDATION);
+    }
+
+    public ConfigField(final String key, final String label, final String type, final boolean required, final boolean sensitive, final BiFunction<FieldValueModel, FieldModel, Collection<String>> validationFunction) {
+        this(key, label, type, required, sensitive, FieldGroup.DEFAULT, "", validationFunction);
+    }
+
+    public Collection<String> validate(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        return validate(fieldToValidate, fieldModel, List.of(validationFunction));
+    }
```

Comment:
We should have an abstract method that is ran after we determine if the field should be required and after we run the passed function. This would help force a contract with child classes